File: app/src/main/java/com/example/sqlite/room/AIQuestionEntity.java
package com.example.sqlite.room;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.ColumnInfo;
import androidx.room.ForeignKey;

@Entity(tableName = "ai_questions",
        foreignKeys = @ForeignKey(entity = UserEntity.class,
                                  parentColumns = "user_id",
                                  childColumns = "user_id",
                                  onDelete = ForeignKey.CASCADE))
public class AIQuestionEntity {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ai_question_id")
    public long aiQuestionId;

    @ColumnInfo(name = "user_id")
    public long userId;

    @ColumnInfo(name = "question_text")
    public String questionText;

    @ColumnInfo(name = "image_uri")
    public String imageUri;

    @ColumnInfo(name = "image_description")
    public String imageDescription;

    @ColumnInfo(name = "ai_answer")
    public String aiAnswer;

    @ColumnInfo(name = "status")
    public String status;

    @ColumnInfo(name = "created_at")
    public String createdAt;
}

File: app/src/main/java/com/example/sqlite/room/AIQuestionEntity.kt
package com.example.sqlite.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "ai_questions",
    foreignKeys = [
        ForeignKey(
            entity = UserEntity::class,
            parentColumns = ["user_id"],
            childColumns = ["user_id"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["user_id"])]
)
data class AIQuestionEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ai_question_id")
    var aiQuestionId: Long = 0,

    @ColumnInfo(name = "user_id")
    var userId: Long = 0,

    @ColumnInfo(name = "question_text")
    var questionText: String = "",

    @ColumnInfo(name = "image_uri")
    var imageUri: String? = null,

    @ColumnInfo(name = "image_description")
    var imageDescription: String? = null,

    @ColumnInfo(name = "ai_answer")
    var aiAnswer: String = "",

    @ColumnInfo(name = "status")
    var status: String? = null,

    @ColumnInfo(name = "created_at")
    var createdAt: String? = null
)
